Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -207,12 +207,28 @@ def check_shim_references(prefix)
     keg = Keg.new(prefix)
 
     matches = []
-    keg.each_unique_file_matching(HOMEBREW_SHIMS_PATH) do |f|
-      match = f.relative_path_from(keg.to_path)
+    keg.each_unique_file_matching(HOMEBREW_SHIMS_PATH) do |file|
+      relative_file = file.relative_path_from(keg.to_path)
 
-      next if match.to_s.match? %r{^share/doc/.+?/INFO_BIN$}
+      next if relative_file.to_s.match? %r{^share/doc/.+?/INFO_BIN$}
 
-      matches << match
+      result = if file.binary_executable? || file.dylib?
+        Utils.popen_read("strings", "-", file.to_s) do |io|
```

Comment:
If this logic is essentially shared with `keg_contain?` can they both call a shared function?